### ISSUE TITLE
Focus issue in Sheet component

### ISSUE THREAD

**Issue by Ro-Matos** [Telefonica/mistica-design#2532]
### Describe the bug

When the Sheet component is open, the content of the underlying page/layer remains navigable and focusable.
This includes focus navigation via keyboard and screen readers.
The issue occurs on both Mobile and Desktop environments.

Evidences

https://github.com/user-attachments/assets/30489a9f-3463-4912-913d-92ec2da40267

The issue occurs in all variations:

https://github.com/user-attachments/assets/70f7e4e2-0900-4d31-9fe1-a42e46ca5bfb

### What libraries are you seeing the problem on?

Mobile Library, Desktop Library

### Steps to reproduce

Navigate with the keyboard using the TAB key.

### Expected behavior

When the Sheet is open:

The background content must not be focusable or navigable.
A _focus-trap_ should be active, keeping keyboard and assistive technology focus restricted to the open Sheet.
Focus should remain inside the Sheet until it is closed.

**Comment by aweell** [comment]
We have open: https://jira.tid.es/browse/WEB-2388 to review this issue, thanks for reporting!
